keras_nlp/tokenizers/word_piece_tokenizer_trainer.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Trainer for Word Piece Tokenizer."""
+
+import tensorflow as tf
+import tensorflow_text as tf_text
+from tensorflow_text.tools.wordpiece_vocab import (
+    wordpiece_tokenizer_learner_lib as learner,
+)
+
+from keras_nlp.tokenizers.word_piece_tokenizer import PUNCTUATION_REGEX
+from keras_nlp.tokenizers.word_piece_tokenizer import (
+    WHITESPACE_AND_PUNCTUATION_REGEX,
+)
+
+
+def compute_word_piece_vocabulary(
+    data,
+    vocabulary_size,
+    vocabulary_output_file=None,
+    lowercase=True,
+    strip_accents=True,
+    split=True,
+    suffix_indicator="##",
+    reserved_tokens=["[PAD]", "[CLS]", "[SEP]", "[UNK]", "[MASK]"],
+):
+    """A utility to train a Word Piece vocabulary.
+
+    This function can be used to train a Word Piece vocabulary from an input
+    dataset, or a list of filenames.
+
+    Args:
+        data: A tf.data.Dataset, or a list of filenames.
+        vocabulary_size: The maximum size of a vocabulary to be trained.
+        vocabulary_output_file: The location to write a vocabulary file.
+        lowercase: If true, the input text will be first lowered before
+            tokenization.
+        strip_accents: If true, all accent marks will be removed from text
+            before tokenization.
+        split: If true, input will be split on whitespace and punctuation
+            marks, and all punctuation marks will be kept as tokens. If false,
+            input should be split ("pre-tokenized") before calling the
+            tokenizer, and passed as a dense or ragged tensor of whole words.
+            `split` is required to be`True` when `data` is a list of filenames.
+        suffix_indicator: The characters prepended to a wordpiece to indicate
+            that it is a suffix to another subword.
+        reserved_tokens: A list of tokens that must be included in the vocabulary.
+
+    Returns:
```

Comment:
Let's add some usage examples!

Maybe
 - Simple tf.data.Dataset.from_tensor_slices example, showing creating a tokenizer and mapping over it.
 - A custom splitting example, similar to the one I shared in the colab, showing mapping over the dataset for custom normalization, then training a vocab, then making a tokenizer and mapping again.

Suggest, this as a custom split rule to show. Stripping punctuation is a legit use case:

```
def normalize_and_split(x):
  """Strip punctuation and split on whitespace"""
  x = tf.strings.regex_replace(x, r"\p{P}", "")
  return tf.strings.split(x)
```

Reply:
added examples